lib/api/completeMultipartUpload.js

```diff
@@ -402,20 +434,26 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                     return next(null, destBucket, dataLocations,
                         metaStoreParams, mpuBucket, mpuOverviewKey,
                         aggregateETag, storedPartsAsObjects, objMD,
-                        extraPartLocations, pseudoCipherBundle, dataToDelete);
+                        extraPartLocations, pseudoCipherBundle, dataToDelete,
+                        newObjData);
                 });
         },
         function storeAsNewObj(destinationBucket, dataLocations,
             metaStoreParams, mpuBucket, mpuOverviewKey, aggregateETag,
             storedPartsAsObjects, objMD, extraPartLocations, pseudoCipherBundle,
-            dataToDelete, next) {
+            dataToDelete, newObjData, next) {
             return services.metadataStoreObject(destinationBucket.getName(),
                 dataLocations, pseudoCipherBundle, metaStoreParams,
                 (err, res) => {
                     if (err) {
                         return next(err, destinationBucket);
                     }
-                    const generatedVersionId = res ? res.versionId : undefined;
+                    let generatedVersionId;
+                    if (newObjData) {
+                        generatedVersionId = newObjData.VersionId || undefined;
```

Comment:
yeah, I guess this might not work with other backends. I'll add some processing of the object returned from AWS in the gateway